sonic-mgmt test changes needed for FS voq

lakshmi-nexthop opened the pull request:
Summary:
Fixes # (issue)

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
    - [ ] Skipped for non-supported platforms
- [ x] Test case improvement


### Back port request
- [ ] 202205
- [ ] 202305
- [ ] 202311
- [ ] 202405
- [ ] 202411
- [ ] 202505

### Approach
#### What is the motivation for this PR?
- sonic-mgmt tests need to be improved for FS voq

#### How did you do it?
- access to Chassis DB is done only if the voq is a chassis

#### How did you verify/test it?
- ran the test on FS voq

#### Any platform specific information?

#### Supported testbed topology if it's a new test case?

### Documentation

mssonicbld commented:
/azp run

arlakshm commented:
@lakshmi-nexthop, please resolve the conflicts

venu-nexthop commented on ansible/roles/test/files/ptftests/py3/hash_test.py:
```diff
@@ -261,9 +261,10 @@ def _get_ip_proto(self, ipv6=False):
         # ip_proto 254 is experimental
         # MLNX ASIC can't forward ip_proto 254, BRCM is OK, skip for all for simplicity
-        skip_protos = [2, 253, 4, 41, 60, 254]
+        skip_protos = [2, 253, 4, 41, 51, 60, 254]
```
> This was a bug on Q3D from BRCM. They've fixed it in `REL_SDK_HSDK_SAI_13.2.1.9_SUG_101725` CS00012424643
> 
> Are you still seeing failures with BRCM's fix?

It is working now. Will remove this change. Thanks for reviewing!

mssonicbld commented:
/azp run

vmittal-msft commented:
@yxieca please help merge.

tjchadaga commented on tests/common/fixtures/fib_utils.py:
```diff
@@ -44,16 +44,18 @@ def get_t2_fib_info(duthosts, duts_cfg_facts, duts_mg_facts, testname=None):
                     dut_inband_intfs.setdefault(duthost.hostname, []).extend(asic_cfg_facts[1]['VOQ_INBAND_INTERFACE'])
             dut_port_channels.setdefault(duthost.hostname, {}).update(asic_cfg_facts[1].get('PORTCHANNEL_MEMBER', {}))
     sys_neigh = {}
-    if switch_type == "voq":
```
wouldn't this section get skipped for multi-asic pizzabox as well now?

rlhui replied:
@lakshmi-nexthop could you please check/respond on this? thanks.

lakshmi-nexthop replied:
I think this is a valid concern. Below line
is_chassis = duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_chassis")
should be 
is_chassis = duthost.dut_basic_facts()['ansible_facts']['dut_basic_facts'].get("is_voq_chassis")

Which covers chassis and multi asic cases.

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

lakshmi-nexthop commented on tests/common/plugins/conditional_mark/tests_mark_conditions_voq.yaml:
```diff
@@ -0,0 +1,29 @@
+  skip:
+    reason: "Skipped as the test applies to chassis only"
+    conditions:
+      - "is_chassis==False"
```
same as above.

lakshmi-nexthop commented on tests/common/plugins/conditional_mark/tests_mark_conditions_voq.yaml:
```diff
@@ -0,0 +1,29 @@
+  skip:
+    reason: "Skipped as the test applies to chassis only"
+    conditions:
+      - "is_chassis==False"
```
same as above.

lakshmi-nexthop commented:
> > @lakshmi-nexthop This change is breaking the T2 testbed, causing the T2 kvm test to fail
> > ```
> > 
> > fib/test_fib.py::test_basic_fib[True-True-1514] 
> > -------------------------------- live log call ---------------------------------
> > 19/01/2026 23:50:55 __init__.pytest_runtest_call             L0040 ERROR  | Traceback (most recent call last):
> >   File "/opt/venv/lib/python3.12/site-packages/_pytest/python.py", line 1720, in runtest
> >     self.ihook.pytest_pyfunc_call(pyfuncitem=self)
> >   File "/opt/venv/lib/python3.12/site-packages/pluggy/_hooks.py", line 512, in __call__
> >     return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)
> >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/opt/venv/lib/python3.12/site-packages/pluggy/_manager.py", line 120, in _hookexec
> >     return self._inner_hookexec(hook_name, methods, kwargs, firstresult)
> >            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/opt/venv/lib/python3.12/site-packages/pluggy/_callers.py", line 167, in _multicall
> >     raise exception
> >   File "/opt/venv/lib/python3.12/site-packages/pluggy/_callers.py", line 121, in _multicall
> >     res = hook_impl.function(*args)
> >           ^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/opt/venv/lib/python3.12/site-packages/_pytest/python.py", line 166, in pytest_pyfunc_call
> >     result = testfunction(**testargs)
> >              ^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/var/src/sonic-mgmt/tests/fib/test_fib.py", line 300, in test_basic_fib
> >     fib_files = fib_info_files_per_function(duthosts, ptfhost, duts_running_config_facts, duts_minigraph_facts,
> >                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/var/src/sonic-mgmt/tests/fib/test_fib.py", line 255, in fib_info_files_per_function
> >     fib_info = get_t2_fib_info(duthosts, duts_config_facts, duts_minigraph_facts, testname)
> >                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "/var/src/sonic-mgmt/tests/common/fixtures/fib_utils.py", line 118, in get_t2_fib_info
> >     remote_duthost_name = sys_neigh[neigh_ip]['duthost_name']
> >                           ~~~~~~~~~^^^^^^^^^^
> > KeyError: '10.0.0.7'
> > 
> > FAILED        
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Please check
> 
> I am testing on kvm to repro the issue above.

@arlakshm I was able to reproduce the problem above with t2 kvm. 
Fixed the issue and updated the diff in PR. Also tested in t2 KVM, I no longer see the problem. Pl proceed with your testing with latest diff.

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

mssonicbld commented:
/azp run

mssonicbld commented:
@lakshmi-nexthop PR conflicts with 202511 branch
